consensus/src/state_computer.rs

```diff
@@ -124,48 +125,54 @@ impl StateComputer for ExecutionProxy {
         let payload_manager = self.payload_manager.lock().as_ref().unwrap().clone();
         let txn_deduper = self.transaction_deduper.lock().as_ref().unwrap().clone();
         let txn_shuffler = self.transaction_shuffler.lock().as_ref().unwrap().clone();
-        let txns = payload_manager.get_transactions(block).await?;
+        let txn_notifier = self.txn_notifier.clone();
+        let txns = match payload_manager.get_transactions(block).await {
+            Ok(txns) => txns,
+            Err(err) => return Box::pin(async move { Err(err) }),
+        };
 
         let deduped_txns = txn_deduper.dedup(txns);
         let shuffled_txns = txn_shuffler.shuffle(deduped_txns);
 
-        let block_gas_limit = *self.maybe_block_gas_limit.lock();
+        let maybe_block_gas_limit = *self.maybe_block_gas_limit.lock();
 
         // TODO: figure out error handling for the prologue txn
-        let executor = self.executor.clone();
-
+        let timestamp = block.timestamp_usecs();
         let transactions_to_execute = block.transactions_to_execute(
             &self.validators.lock(),
             shuffled_txns.clone(),
-            block_gas_limit,
+            maybe_block_gas_limit,
         );
 
-        let compute_result = monitor!(
-            "execute_block",
-            tokio::task::spawn_blocking(move || {
-                executor.execute_block(
-                    (block_id, transactions_to_execute).into(),
-                    parent_block_id,
-                    block_gas_limit,
-                )
-            })
-            .await
-        )
-        .expect("spawn_blocking failed")?;
-
-        observe_block(block.timestamp_usecs(), BlockStage::EXECUTED);
+        let fut = self
+            .execution_pipeline
+            .queue(
+                (block_id, transactions_to_execute).into(),
+                parent_block_id,
+                maybe_block_gas_limit,
+            )
+            .await;
 
-        // notify mempool about failed transaction
-        if let Err(e) = self
-            .txn_notifier
-            .notify_failed_txn(shuffled_txns, &compute_result)
-            .await
-        {
-            error!(
-                error = ?e, "Failed to notify mempool of rejected txns",
+        Box::pin(async move {
+            debug!(
+                block_id = %block_id,
```

Comment:
why %? that prints out the whole 32 bytes

Reply:
-_- looked nicer to me. But you are the king.